### PR TITLE
CRM-20385 (IIDA-72), fixed deferred revenue account validation

### PR DESCRIPTION
**Pull request by pradpnayak**
https://issues.civicrm.org/jira/browse/CRM-20385

### CODE REVIEW

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -193,8 +193,7 @@ public static function add(&$params, $ids = array()) {
       $params['prevContribution'] = self::getOriginalContribution($contributionID);
     }
 
-    // CRM-16189
-    CRM_Financial_BAO_FinancialAccount::checkFinancialTypeHasDeferred($params, $contributionID);
+    CRM_Core_BAO_FinancialTrxn::generateRevenueRecognitionDate($params, $contributionID);
```
Comment (monishdeb) [inline CRM/Contribute/BAO/Contribution.php]: @pradpnayak I encountered a fatal error while doing backoffice contribution. 
```
PHP Fatal error:  Call to undefined method CRM_Core_BAO_FinancialTrxn::generateRevenueRecognitionDate() in /Users/monish/src/civicrm/CRM/Contribute/BAO/Contribution.php on line 196
```
didn't find any definition of it in core or is there any other dependent PR for this?

**Comment by monishdeb** [comment]
@pradpnayak I have made some additional fixes https://github.com/civicrm/civicrm-core/pull/10114/commits/0adcec25099013b66651c1d0030dd320146fec2e 

Apart from that there is only issue I have reported on the PR, please have a look

**Comment by pradpnayak** [comment]
@monishdeb i have fixed the Fatal error and also modified your changes to remove extra condition.
I have also squashed the commit.

**Comment by monishdeb** [comment]
Thanks @pradpnayak, will test the final patch.

**Comment by monishdeb** [comment]
@pradpnayak the changes look good, however found a test failure due to change in validation message. Fixed it here https://github.com/civicrm/civicrm-core/pull/10114/files#diff-90c98c96619663881a4653ce51481efbR334 

@eileenmcnaughton I have tested the PR and commented my feedback here https://issues.civicrm.org/jira/browse/CRM-20385?focusedCommentId=102208&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-102208 , can you merge it if the final change looks good to ya ?

**Comment by colemanw** [comment]
@eileenmcnaughton can you review this please?

**Comment by eileenmcnaughton** [comment]
Yeah this looks like a code simplicification & the review notes from Monish & issue description from pradeep seem clear - merging

**Pull request merged**